### PR TITLE
Functionality to correct the parser path

### PR DESCRIPTION
**Pull request by StanleyGoldman**
- The `Parser` needs to collect a Dictionary of projects and Dictionary of compilation targets and their paths relative to it's project
- The `Submitter` needs to know the root where the solution was cloned to.

### CODE REVIEW

File: MSBLOC.Core/Model/ProjectDetails.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        private string GetClonePath(string itemProjectPath)
+        {
+            return Path.Combine(ProjectDirectory, itemProjectPath)
+                .Split(new[] {CloneRoot}, StringSplitOptions.RemoveEmptyEntries)
+                .First();
+        }
+
+        public string GetPath(string itemProjectPath)
+        {
+            return _paths[itemProjectPath];
```
Comment (profet23) [inline MSBLOC.Core/Model/ProjectDetails.cs]: Calling `GetPath` with an `itemProjectPath` that doesn't exist will throw a `KeyNotFoundException` which doesn't make a lot of sense to the user. Should probably test if key exists and throw a custom exception.

File: MSBLOC.Core/Services/Submitter.cs [hunk 2]
```diff
@@ -22,26 +23,31 @@ public Submitter(ICheckRunsClient checkRunsClient, ILogger<Submitter> logger = n
...
+                    warningLevel = CheckWarningLevel.Warning;
+                }
+                else if (annotation.AnnotationWarningLevel == AnnotationWarningLevel.Notice)
+                {
+                    warningLevel = CheckWarningLevel.Notice;
+                }
+                else
+                {
+                    warningLevel = CheckWarningLevel.Failure;
+                }
```
Comment (profet23) [inline MSBLOC.Core/Services/Submitter.cs]: Switch statement?

File: MSBLOC.Core/Services/Submitter.cs [hunk 3]
```diff
@@ -54,15 +60,15 @@ public Submitter(ICheckRunsClient checkRunsClient, ILogger<Submitter> logger = n
...
+        private static NewCheckRunAnnotation NewCheckRunAnnotation(string filename, int lineNumber,
             int endLine, CheckWarningLevel checkWarningLevel, string message, string title)
         {
             if (endLine == 0)
             {
                 endLine = lineNumber;
             }
 
-            return new NewCheckRunAnnotation(filename, blobHref, lineNumber,
+            return new NewCheckRunAnnotation(filename, "", lineNumber,
```
Comment (profet23) [inline MSBLOC.Core/Services/Submitter.cs]: `string.Empty`

File: MSBLOC.Core/Interfaces/ISubmitter.cs [hunk 4]
```diff
@@ -7,6 +7,6 @@ namespace MSBLOC.Core.Interfaces
     public interface ISubmitter
     {
         Task<CheckRun> SubmitCheckRun(string owner, string name, string headSha,
-            string checkRunName, ParsedBinaryLog parsedBinaryLog, string checkRunTitle, string checkRunSummary);
+            string checkRunName, BuildDetails buildDetails, string checkRunTitle, string checkRunSummary, string cloneRoot);
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply MSBLOC.Core/Interfaces/ISubmitter.cs]: I'll address that in #8

File: MSBLOC.Core/Model/BuildDetails.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using MSBLOC.Core.Services;
+
+namespace MSBLOC.Core.Model
+{
+    public class BuildDetails
+    {
+        public List<Annotation> Annotations { get; }
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply MSBLOC.Core/Model/BuildDetails.cs]: 👍

File: MSBLOC.Core/Model/BuildDetails.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            EndLine = endLine;
+        }
+    }
+
+    public enum AnnotationWarningLevel
+    {
+        Notice,
+        Warning,
+        Failure,
+    }
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply MSBLOC.Core/Model/BuildDetails.cs]: 👍

File: MSBLOC.Core/Model/ProjectDetails.cs (hunk 1 above)
Comment (StanleyGoldman) [reply MSBLOC.Core/Model/ProjectDetails.cs]: 👍

File: MSBLOC.Core/Services/Submitter.cs (hunk 2 above)
Comment (StanleyGoldman) [reply MSBLOC.Core/Services/Submitter.cs]: 👍

File: MSBLOC.Core/Services/Submitter.cs (hunk 3 above)
Comment (StanleyGoldman) [reply MSBLOC.Core/Services/Submitter.cs]: Addressed in #8

File: MSBLOC.Core/Model/ProjectDetails.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            ProjectFile = projectFile;
+
+            ProjectDirectory = Path.GetDirectoryName(projectFile)
+                               ?? throw new InvalidOperationException(
+                                   "Path.GetDirectoryName(startedEventArgs.ProjectFile) is null");
+
+            _paths = new Dictionary<string, string>();
+        }
+
+        public void AddItems(params string[] itemProjectPaths)
```
[the remark this answers is not in the document]
Comment (StanleyGoldman) [reply MSBLOC.Core/Model/ProjectDetails.cs]: I'll trade you `ReplaceItems` for `AddOrReplaceItems`

File: MSBLOC.Core/Model/ProjectDetails.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+using System.Linq;
+
+namespace MSBLOC.Core.Model
+{
+    public class ProjectDetails
+    {
+        public string ProjectFile { get; }
+        public string CloneRoot { get; }
+        public string ProjectDirectory { get; }
+        public IReadOnlyDictionary<string, string> Paths => _paths;
```
Comment (profet23) [inline MSBLOC.Core/Model/ProjectDetails.cs]: Shouldn't just be a cast

`public IReadOnlyDictionary<string, string> Paths => new ReadOnlyDictionary<string, string>(_paths);`

**Pull request merged**